### ISSUE TITLE
Send Transport "produce" listener not running on .produce()

### ISSUE THREAD

**Issue by Gerald-97** [Blancduman/mediasoup-client-flutter#85]
Hello @Blancduman, good day. Thank you for all your work and guidance here. It's an immense help to the community.

I'm also new to mediasoup and I'm not sure what I'm doing wrong.
I'm running the `sendTransport.produce()` according to the docs but the `"produce"` listener doesn't run.

Here's a sample of my code.

[Gist implementation](https://gist.github.com/Gerald-97/226fd9aab86c8a8bd2d3e202c9243f45)

The `sendTransport.on("connect",` listener runs but the `sendTransport.on("produce",` doesn't, and there are no errors on create sendTransport or Device.

**Comment by appalareddi** [comment]
@Gerald-97 
I think you're supposed pass back the producer id that you got from your server. 

```dart
sendTransport.on("produce", (data, cb, errCb) async {
        socket.emitWithAck("transport-produce", {
          "kind": data["kind"],
          "rtpParameters": data["rtpParameters"],
          "appData": data["appData"]
        }, ack: (id, producersExist) {
          data['callback'](id); // this will make the future to resolve with your producer id
        });
      });
```

**Comment by Gerald-97** [comment]
I get that part, but the issue I'm stating is, I cannot listen to the .on("produce"). The "produce" listener is not firing. So I can't access the `id` at all.
The `sendTransport.on("produce", (data, cb, errCb) async {` is not running.

**Comment by appalareddi** [comment]
@Gerald-97 
I think the listener functions are supposed to take only one parameter, which is `data`, this will have `callback`, `errback` values.

Try calling that `data['callback]()` in `connect` listener also and see if producer is getting fired. I also faced this same issue, once I called the `callback` function which i am getting in `data`, `produce` listener got fired.

```dart
sendTransport.on("connect", (data, cb, errCb) async {
        socket.emitWithAck("transport-connect",
            {"dtlsParameters": data["dtlsParameters"].toMap()}, ack: data['callback']);
      });
```
